packages/fuels/src/cli/utils/getLatestFuelsVersion.test.ts

```diff
@@ -0,0 +1,50 @@
+import * as cacheMod from './fuelsVersionCache';
+import { getLatestFuelsVersion } from './getLatestFuelsVersion';
+
+/**
+ * @group node
+ */
+describe('getLatestFuelsVersion', () => {
+  beforeEach(() => {
+    vi.resetAllMocks();
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  it('should fail if fetch fails', async () => {
+    vi.spyOn(global, 'fetch').mockImplementation(() =>
+      Promise.reject(new Error('Failed to fetch'))
+    );
+    await expect(getLatestFuelsVersion()).rejects.toThrowError('Failed to fetch');
+  });
+
+  it('should throw if fetch times out', async () => {
+    vi.spyOn(global, 'fetch').mockImplementation(
+      () =>
+        new Promise((resolve) => {
+          setTimeout(resolve, 5000);
+        })
+    );
+    await expect(getLatestFuelsVersion()).rejects.toThrow();
+  });
+
+  it('should return cached version if it exists', async () => {
+    const cachedVersion = '1.0.0';
+    vi.spyOn(cacheMod, 'checkAndLoadCache').mockReturnValue(cachedVersion);
+    const result = await getLatestFuelsVersion();
+    expect(result).toEqual('1.0.0');
+  });
+
+  it('should fetch if there is no cache or the cache is expired', async () => {
+    const mockResponse = new Response(JSON.stringify({ version: '1.0.0' }));
+    const fetchSpy = vi.spyOn(global, 'fetch').mockReturnValue(Promise.resolve(mockResponse));
+    const saveCacheSpy = vi.spyOn(cacheMod, 'saveToCache').mockImplementation(() => {});
+    vi.spyOn(cacheMod, 'checkAndLoadCache').mockReturnValue(null);
+    const version = await getLatestFuelsVersion();
+    expect(fetchSpy).toHaveBeenCalled();
+    expect(version).toEqual('1.0.0');
+    expect(saveCacheSpy).toHaveBeenCalledWith('1.0.0');
```

Comment:
Because [I'm presuming] you're mocking everything related to the cache, the tests may not cover some critical parts of the functionality. Can we get all files in 100%?

![coverage](https://github.com/user-attachments/assets/ed27f03a-4b1f-46cc-8cab-c295d85111b2)

Reply:
Think this [relates](https://github.com/FuelLabs/fuels-ts/pull/3186#discussion_r1787510922) - we should cover both success and failure conditions in my opinion.